src/extension.ts

```diff
@@ -501,6 +533,32 @@ function is<T extends object>(o: object, propOrMatcher?: keyof T | ((o: any) =>
   return value === undefined ? (o as any)[propOrMatcher] !== undefined : (o as any)[propOrMatcher] === value;
 }
 
+async function isValidTerraformFile(file: vscode.TextDocument): Promise<boolean> {
+  const filename = file.uri.path;
+  const isTfFile = /.*\.tf$/.test(filename);
+
+  if (!isTfFile) {
+    debugLog.appendLine(`debug: ${filename} is not a valid terraform file extension`);
+    return false;
+  }
+
+  const symbols = await commands.executeCommand<SymbolInformation[]>('vscode.executeDocumentSymbolProvider', file.uri);
+  if (symbols === undefined) {
+    debugLog.appendLine(`debug: no valid terraform symbols found for file ${filename}`);
+    return false
+  }
+
+  for (const sym of symbols) {
+    if (sym.name.indexOf('resource') !== -1 || sym.name.indexOf('module') !== -1) {
```

Comment:
Not sure I'm understanding this fully but is it valid to ignore files that don't have modules or resources? Could files that only contain variables also need to trigger a run since the default values of variables can have a cost impact?

Reply:
Yup I was thinking about this last night, you are correct we can't just ignore files without `resource` and `module` blocks at this point as we're rebuilding the whole Infracost estimate. I'll alter it to just ignore files without any symbols (e.g. blank files)

Reply:
removed in https://github.com/infracost/vscode-infracost/pull/20/commits/1ae81fa58db3d00c7f398a09473c36e4f89dfdf0